### ISSUE TITLE
Constructor arguments prefixed with "true"?

### ISSUE THREAD

**Issue by osyrisrblx** [Perryvw/TypescriptToLua#136]
It would appear all constructor arguments get prefixed with a true literal. This is fine for generated classes because they have the logic to deal with this.

However, Classes from Lua become unusable. Because "new" is a reserved keyword in TypeScript, you can't simply do `Class.new(args)`, instead you must do `new Class(args)`, but that is transpiled to `Class.new(true, args)`.

Is there a reason for this behavior? If it's a bug, I'd be happy to make a PR.

**Comment by apemanzilla** [comment]
As a temporary workaround, dos `Class["new"](...)` work?

**Comment by osyrisrblx** [comment]
Yes that seems to work, but not very easy to write.

**Comment by Perryvw** [comment]
Definitely not a bug, there are many different OOP styles in lua and whatever style we choose there will always be some methods left out. This is the method we chose to implement OOP. We have not seen complaints for the lifetime of this project, but I'm open to suggestions on how to improve it.

**Comment by osyrisrblx** [comment]
That's fair but what's the use case for the `true` value? Is there a scenario where it isn't `true` for a constructor?

**Comment by Perryvw** [comment]
Creating a subclass is done by creating an instance of the superclass without executing its constructor (`false` in new) and then executing the constructor of the extending class.

**Comment by osyrisrblx** [comment]
Ah that makes more sense.

I usually have seen it done as:
```Lua
local subClass = setmetatable({}, superClass)
subClass.__index = subClass
function subClass.new(...)
    local self = setmetatable({}, subClass)
    -- set values
    return self
end
```

but that is by no means the "correct" way.

Anyways, thank you for making this! I've been messing around with a fork of the repo that allows for compatibility with making games on [Roblox](https://www.roblox.com/).

https://github.com/osyrisrblx/TypescriptToLua/blob/master/src/targets/Transpiler.Rbx.ts

**Comment by Perryvw** [comment]
That's nice, any specific reason you need to change the transpiler instead of just creating declarations files?

**Comment by osyrisrblx** [comment]
Roblox has dramatically changed Lua 5.1 unfortunately. For example "require" is used on game objects instead of strings. So in order to get that compatible with TypeScript's import statements I've had to change that functionality.

Also it doesn't support ambient variable declarations in modules (like how the current runtime library is setup). So I had to change that to be a table of exports.

Pretty minor stuff.
